Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -758,28 +758,40 @@ namespace OpenInfraPlatform {
 					return object_placement_matrix;
                 }
 				
-				std::vector<carve::geom::vector<2>> convertifcGridAxis(const EXPRESSReference<typename IfcEntityTypesT::IfcGridAxis>& gridAxis) const throw(...){
-					
-					std::vector<carve::geom::vector<2>> Axis;
-					std::vector<carve::geom::vector<2>> SegmentStartPoints;
+				std::tuple<std::vector<carve::geom::vector<2>>,bool> convertIfcGridAxis(const EXPRESSReference<typename IfcEntityTypesT::IfcGridAxis>& gridAxis) const throw(...) {
+
+					std::vector<carve::geom::vector<2>> axis;
+					std::vector<carve::geom::vector<2>> segmentStartPoints;
+					bool sameSense;
+					if (gridAxis->SameSense) {
+						sameSense = true;
+					}
+					else {
+					sameSense = false;
+					}

```

Comment:
I had something else in mind.

You don't need to return the `sameSense` parameter, as you can apply it within this function already. After the `gridConv.convert...`, you could do the following:

```
if( ! gridAxis->SameSense )
    // turn around the axis
    std::reverse( std::begin(axis), std::end(axis) );
```

In this way, the axis "runs" in the other direction, applying the `SameSense` attribute, so that the calculation of offsets in the calling function happen correctly (at least I think so).

Reply:
Resolved: a619817